### ISSUE TITLE
Tried out f-droid android 2.8.0 build, not good

### ISSUE THREAD

**Issue by Fixer-007** [shlomif/PySolFC#263]
I have large 6" android smartphone, my experience is not good so far:

1. Half of the screen is 'About PySol' window, opens every time (why?), it does not care if in portrait mode, it will squeeze it in tiny strip.
2. Absolutely tiny cards (half of the huge screen at best), better in landscape mode obviously, but cards still too small, resizing is available in newer version, but not in this build I guess
4. Menus are not intuitive, figured out how to close them by luck
5. You can open menus near menus

This build is probably by third parties but UI is clearly lacking and just not good to play overall sadly, quite antimobile.

IMO those are the biggest issues you face as first time player. For casuals it is just not ready to play comfortably.

Build: https://f-droid.org/packages/org.lufebe16.pysolfc/

**Comment by joeraz** [comment]
The about PySol window can be disabled through the options menu - startup splash screen option.

As for the Android version in general, main issue right now is that my ability to support the Android version is currently quite limited.  When I took over handling the releases, I don't think I ever got the details on how to actually package and release new versions on f droid.  I do have the ability to make modifications to the Android code, but I can only test it through Kivy's emulator, which isn't quite the same as running on an actual Android device.

I can try to clean up the menu design, but I'm afraid I'd need some additional support with generating builds and testing for it to amount to much.

**Comment by Fixer-007** [comment]
Thank you, to me resizing was the worst problem. Sadly i'm not a programmer, but just in case if you need to test some new builds, I can try.

**Comment by joeraz** [comment]
I'd need to figure out how the Android builds would be generated first.

**Comment by BloodRaven0** [comment]
Just to add to the list of problems, it crashed midway through my first klondike game. Android 10. If you'd like to I can try to make it crash again with logcat running and send you the logs.
It's a real shame there is no polished FOSS solitaire app for android currently, but this one comes close, it just needs some work.

**Comment by joeraz** [comment]
@BloodRaven0 Yes, if you can send me logs or detailed steps for reproduction, I can take a closer look.

Would love to give the Android version more love, but as I don't currently have instructions on how to prepare Android release packages, and can only test through the Kivy debugger, my ability to support it is currently limited.  There have also been recent requests for an iOS version (see #266).  There is definitely potential to create something great here, but I'd need some additional support to make it happen.

**Comment by lufebe16** [comment]
Just placed a new actual version (as of 2.18.0) on f-droid.
Enjoy!

**Comment by Fixer-007** [comment]
Tried 2.18.0, did not change much, same problems.
Does not resize card deck properly, cards are too small, only use like 50% of the screen without centering (ofc depending on game).

**Comment by joeraz** [comment]
Well, if there are new release instructions, I'll review those, and this should hopefully improve my ability to support the Android version going forward.  Though I'm still limited to my Kivy emulator for testing - there, I can see the centering and menu issues there, but not the card size issue described.  It looks like the cards scale based on the size of the game and the size of the screen.  Interestingly, it looks like the Android version scales the text with the cards while the PC/Linux version does not.

I don't think I can fully polish the mobile version on my own, but with a little more research into Kivy, I should be able to resolve some of those items for the next release.

**Comment by lufebe16** [comment]
Currently reviewing the kivy version, which implements the android versions surface. In the last 2-3 years nothing has been changed from that side.

In the kivy version cards are placed on the playground as defined in the game definition. The resulting field depends from the card size and is then scaled into the available Window, so that either width or height matches to the more tight window dimension. The game is not centered within the larger of the two dimensions, but placed top or left (is this the 'centering issue' addressed?). The layout is never touched.

Text and decorating images are placed on the (virtual) card field, not directly on the screen. So they are scaled too (and may get hardly visible)

In work are scaling issues with fonts and decorating images. Coming soon.

**Comment by joeraz** [comment]
Yes, I think those are the centering issues.  It was previously the same way in the desktop version, but last year, I added a "Center layout" option which centers the game layout on the screen based on the larger window dimension.  That works on the tkInter desktop version, but I never got around to porting the option to Kivy.

Aside from that, any changes I made to the Kivy version were pretty much making sure I didn't accidentally break anything there, and exposing whatever new features I added on the desktop side that were relevant.
